Parser/Node.cs

```diff
@@ -16,11 +16,18 @@ public record NProgram (Token Name, NBlock Block) : Node {
 
 // A block contains declarations and a body
 public record NBlock (NDeclarations Declarations, NCompoundStmt Body) : Node {
+   /// <summary>If this block is inside a function, it contains the parameters of the function</summary>
+   public NVarDecl[] VarDecls { get; set; } = Array.Empty<NVarDecl> ();

```

Comment:
This is not very clean. The list of variables is already there in the enclosing NFnDecl node, so having to repeat it here is not good. How about this: when the TypeAnalyzer sees a NFnDecl node, it creates a new symbol table and adds all the parameters in first (before even parsing the declarations section), then you don't need this I think.